src/Device/Driver/OpenVario.cpp

```diff
@@ -46,65 +46,147 @@ class OpenVarioDevice : public AbstractDevice {
   bool PutBugs(double bugs, OperationEnvironment &env) override;
   void OnCalculatedUpdate(const MoreData &basic,
                   const DerivedInfo &calculated) override;
-  static bool POV(NMEAInputLine &line, NMEAInfo &info);
+private:
+  bool POV(NMEAInputLine &line, NMEAInfo &info);
+  bool ComposeWrite(const char p_type,double a, double b, double c,
+  		  OperationEnvironment &env);
+  bool PutIdealPolar(bool force, const DerivedInfo *calculated,
+  		  OperationEnvironment &env);
+  bool PutRealPolar(bool force, const DerivedInfo *calculated,
+  		  OperationEnvironment &env);
+  bool RepeatBugs(OperationEnvironment &env);
+  bool RepeatBallast(OperationEnvironment &env);
+  bool RepeatMacCready(OperationEnvironment &env);
 };
 
+static const DerivedInfo *_calculated = NULL;
+static double _bugs = 1;
+static double _overload = 1;
+static double _mc = 1;
```

Comment:
I have no idea what this obscure commit does, but these global variables are not acceptable!

Reply:
The static variables remember the previous settings, when e.g. PutMacCready() was called last time. The value will be used when the connected device (after a restart) wants to know what the current value was. It is used in RepeatMacCready.
Similarly for Ballast, Bugs, Polar.
I tried to keep the code self contained as much as I could.

Reply:
You explained what those globals do, but that doesn't make them any better.
Maybe you understand it when you answer this question: who owns the memory pointed to by `_calculated`?
Another question: what if a user has two OpenVarios - who will protect those variables from clashing between the two instances?

Reply:
I admit: I didn't think about multiple OpenVarios.
I need to figure out how to replace _calculated .
Can you provide a hint?